### ISSUE TITLE
Add event quota data to the project and organization endpoints

### ISSUE THREAD

**Issue by safanobu** [getsentry/sentry#51246]
### Data needed from the api
We would like to start using the [Spend Allocation](https://docs.sentry.io/product/accounts/quotas/spend-allocation/) new feature, at the same time we would like to be able to get information around the event quota usage through your [api](https://docs.sentry.io/api):
- quota allocated per project
- quota spent per project
- quota allocated for the organization
- quota spent by the organization

For the time being, the [project](https://docs.sentry.io/api/projects/retrieve-a-project/?original_referrer=https%3A%2F%2Fwww.google.com%2F) and [organization](https://docs.sentry.io/api/organizations/retrieve-an-organization/?original_referrer=https%3A%2F%2Fwww.google.com%2F) endpoints do not provide such data.

### Why we need this data?
We need event quota data to be able to set quota depletion alerts through a custom prometheus exporter.

### Alternative solution
We would also be interested in a way to send quota depletion alerts through the Spend Allocation UI

### Product Area

Projects

**Comment by snigdhas** [comment]
Hi @safanobu, thanks for filing this ticket. I'm adding it to the backlog for the issues team. Please note that Sentry does send [Spend Allocation notifications](https://docs.sentry.io/product/accounts/quotas/spend-allocation/#how-spend-allocation-works) when 80% and 100% of the quota is consumed.

**Comment by safanobu** [comment]
> Hi @safanobu, thanks for filing this ticket. Sentry does send [Spend Allocation notifications](https://docs.sentry.io/product/accounts/quotas/spend-allocation/#how-spend-allocation-works) when 80% and 100% of the quota is consumed.

Thank you for the reply @snigdhas ! So these notifications will be scoped to the members of the Project only?

Another reason we wanted quota data available through the API, is to make that information closer to the monitoring tools our developers are using, which is Grafana. So they are more aware of their Sentry usage.

**Comment by clemuscle** [comment]
Hi,
The issue is still relevent, it can bring an improvement to monitor Sentry:
- quota allocated per project
- quota spent per project
- quota allocated for the organization
- quota spent by the organization

Or else is it possible to have access to the quota allocated and the time period of the subscription?

**Comment by sandipb** [comment]
> Hi @safanobu, thanks for filing this ticket. Sentry does send [Spend Allocation notifications](https://docs.sentry.io/product/accounts/quotas/spend-allocation/#how-spend-allocation-works) when 80% and 100% of the quota is consumed.

A big ask for us is to be able to *predict* quota usage and alert on it. We don't just want to be alerted when we have already exhausted our quota. We want to be alerted when a sudden increase of some code issue is going to cause us to breach quota by the end of the month.

If you can just provide an end point to give us the quota usage and limit, we can use any of the observability tools available to predict a breach of threshold by the end of the month and alert on that.

**Comment by nhsiehgit** [comment]
Hey @sandipb ,

You may also want to look into the spike protection feature.

Currently work on this particular feature has been put on pause as we make changes to our subscription plans and pricing models.
We will definitely keep these feature requests in mind as we iterate and improve the feature!

cc @Dhrumil-Sentry

**Comment by sandipb** [comment]
> Hey @sandipb ,
> 
> You may also want to look into the spike protection feature.

Yeah, I am aware of that feature. We probably already have it on, but doesnt entirely address our requirements - we don't want to throttle our errors if they jump, we are ok with spikes. we just want to know if the representative increase will blow our quota by the end of the month.

> 
> Currently work on this particular feature has been put on pause as we make changes to our subscription plans and 
> pricing models. We will definitely keep these feature requests in mind as we iterate and improve the feature!

That is alright. I found a prometheus exporter that I might be able to use for our needs. Though I am a bit intrigued by plans for overhauling the pricing models. My tweets with you folks yesterday (on the Cron monitor announcement thread) suggested that you people are leaning towards on-demand pricing for everything. Maybe that is what you are alluding to?
In any case, this is not the right forum for that discussion. Sorry :)

**Comment by hampsterx** [comment]
any update here, even with project rate limits and spike protection its still easy to blow through the quota and the quota being monthly is a PITA, being able to monitor it from the API is kind of critical. Also really super annoying is the event quota gets blown which stops session replays even tho its unrelated, GRR.

**Comment by brianthi** [comment]
One thing that hasn't been mentioned in this thread yet is the changes we've made recently to spend notifications. We've since changed that system to allow for configurable thresholds to notify from 10% to 90% consumption for both subscription + reserved volumes and on-demand/pay-as-you-go usage. The only caveat to this is that this notification is only sent to owners and billing members. More details in our docs [here](https://docs.sentry.io/product/alerts/notifications/#spend-notifications). I realize this isn't prediction for when you will run out of quota, but it's a step in that direction. 

We haven't yet prioritized the request in this issue, but will provide an update when we do.
